app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+    const {messageStyle, textStyles} = styles;
+
+    if (!postProps.username) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const oldPurpose = postProps.old_purpose;
+    const newPurpose = postProps.new_purpose;
+    let localeHolder;
+
+    if (postProps.new_purpose) {
+        if (postProps.old_purpose) {
+            localeHolder = {
+                id: 'app.channel.post_update_channel_purpose_message.updated_from',
+                defaultMessage: '{username} updated the channel purpose from: {oldPurpose} to: {newPurpose}',
+            };
+
+            return (
+                <Markdown
+                    baseTextStyle={messageStyle}
+                    onPostPress={onPress}
+                    textStyles={textStyles}
+                    value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+                />
+            );
+        }
+
+        localeHolder = {
+            id: 'app.channel.post_update_channel_purpose_message.updated_to',
+            defaultMessage: '{username} updated the channel purpose to: {newPurpose}',
+        };
+
+        return (
+            <Markdown
+                baseTextStyle={messageStyle}
+                onPostPress={onPress}
+                textStyles={textStyles}
+                value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+            />
+        );
+    } else if (postProps.old_purpose) {
+        localeHolder = {
+            id: 'app.channel.post_update_channel_purpose_message.removed',
+            defaultMessage: '{username} removed the channel purpose (was: {oldPurpose})',
+        };
+
+        return (
+            <Markdown
+                baseTextStyle={messageStyle}
+                onPostPress={onPress}
+                textStyles={textStyles}
+                value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+            />
+        );
+    }
+
+    return null;
+};
+
+const renderDisplayNameChangeMessage = (postBodyProps, styles, intl) => {
+    const {onPress, postProps} = postBodyProps;
+    const {messageStyle, textStyles} = styles;
+    const oldDisplayName = postProps.old_displayname;
+    const newDisplayName = postProps.new_displayname;
+
+    if (!(postProps.username && postProps.old_displayname && postProps.new_displayname)) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const localeHolder = {
+        id: 'api.channel.post_update_channel_displayname_message_and_forget.updated_from',
+        defaultMessage: '{username} updated the channel display name from: {oldDisplayName} to: {newDisplayName}',
+    };
+
+    return (
+        <Markdown
+            baseTextStyle={messageStyle}
+            onPostPress={onPress}
+            textStyles={textStyles}
+            value={formattedMessage(localeHolder, {username, oldDisplayName, newDisplayName}, intl)}
```

Comment:
if you remove the `formattedMessage` don't forget to replace it here with
```suggestion
            value={intl.formatMessage(localeHolder, {username, oldDisplayName, newDisplayName})}
```

Reply:
Done.